From 0eede5695e2078aa2e53ed668106080176f05471 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Thu, 25 Jan 2024 17:27:42 +0000
Subject: [PATCH 646/697] drm/vc4: Initialise the tv_mode property default from
 cmdline_mode.

With the command line parser now providing the information about
the tv mode, use that as the preferred choice for initialising the
default of the tv_mode property.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/gpu/drm/vc4/vc4_vec.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/gpu/drm/vc4/vc4_vec.c
+++ b/drivers/gpu/drm/vc4/vc4_vec.c
@@ -388,7 +388,9 @@ static const struct drm_prop_enum_list l
 static enum drm_connector_tv_mode
 vc4_vec_get_default_mode(struct drm_connector *connector)
 {
-	if (vc4_vec_tv_norm) {
+	if (connector->cmdline_mode.tv_mode_specified) {
+		return connector->cmdline_mode.tv_mode;
+	} else if (vc4_vec_tv_norm) {
 		int ret;
 
 		ret = drm_get_tv_mode_from_name(vc4_vec_tv_norm, strlen(vc4_vec_tv_norm));
